standard_sequence/classy_sequence.py

```diff
@@ -861,263 +318,528 @@ def _do_molasses_tof_sequence(self, t, reset_mot = False):
+                t, bias_field_vector=(op_biasx_field, op_biasy_field, op_biasz_field)
+            )
             # ramp detuning to 4 -> 4, 3 -> 3
-            self.D2Lasers_obj.ramp_ta_freq(t, 0, CONST_TA_PUMPING_DETUNING)
-            self.D2Lasers_obj.ramp_repump_freq(t, 0, CONST_REPUMP_DEPUMPING_DETUNING)
+            self.D2Lasers_obj.ramp_ta_freq(t, D2Lasers.CONST_TA_VCO_RAMP_TIME, CONST_TA_PUMPING_DETUNING)
+            # self.D2Lasers_obj.ramp_repump_freq(t, D2Lasers.CONST_TA_VCO_RAMP_TIME, 0)
+            self.D2Lasers_obj.ramp_repump_freq(t, D2Lasers.CONST_TA_VCO_RAMP_TIME, CONST_REPUMP_DEPUMPING_DETUNING)
             # Do a sigma+ pulse
             # TODO: is shot_globals.op_ramp_delay just extra fudge time? can it be eliminated?
-            t += max(CONST_TA_VCO_RAMP_TIME, shot_globals.op_ramp_delay)
-            t = self.D2Lasers_obj.do_pulse(t - CONST_SHUTTER_TURN_ON_TIME,
-                                       shot_globals.odp_repump_time,
-                                       ShutterConfig.OPTICAL_PUMPING_FULL,
-                                       shot_globals.odp_ta_power,
-                                       shot_globals.odp_repump_power,)
-            # Need to turn off the TA before repump, Sam claims this timing should work
-            assert shot_globals.odp_ta_time > shot_globals.odp_repump_time, "TA time should be longer than repump for depumping to F = 3"
+            t += max(D2Lasers.CONST_TA_VCO_RAMP_TIME, shot_globals.op_ramp_delay)
+            t, t_aom_start = self.D2Lasers_obj.do_pulse(
+                t,
+                shot_globals.odp_ta_time,
+                ShutterConfig.OPTICAL_PUMPING_FULL,
+                shot_globals.odp_ta_power,
+                shot_globals.odp_repump_power,
+                close_all_shutters = close_all_shutters
+            )
+
+            t_aom_off = t_aom_start + shot_globals.odp_ta_time
+
+            assert (
+                shot_globals.odp_ta_time > shot_globals.odp_repump_time
+            ), "TA time should be longer than repump for depumping to F = 3"
             # TODO: test this timing
-            self.D2Lasers_obj.ta_aom_off(t + shot_globals.odp_ta_time - shot_globals.odp_repump_time)
-            # Close the shutters
-            self.D2Lasers_obj.update_shutters(t, ShutterConfig.NONE)
-            t += CONST_SHUTTER_TURN_OFF_TIME
+            self.D2Lasers_obj.repump_aom_off(
+                t_aom_start + shot_globals.odp_repump_time
+            )
 
-            return t
 
-        else:
-            raise NotImplementedError("This optical depumping method is not implemented")
+            return t, t_aom_off
 
+        else:
+            raise NotImplementedError(
+                "This optical depumping method is not implemented"
+            )
 
-    def kill_F4(self, t):
-        ''' Push away atoms in F = 4 '''
+    def kill_F4(self, t, shutter_config = ShutterConfig.OPTICAL_PUMPING_TA, close_all_shutters = True):
```

Comment:
Add explanation for use of shutter_config argument re: speed of pulse after optical pumping. Does this need to be a function argument or can it be hard-coded?

Reply:
No, it doesn't. Change the code to be automatic switch the shutter configuration. If the current shutter config is OPTICAL_PUMPING_FULL, then stay the same to make sure turning on/off the beam with just aom, repump is off using only aom. Else if we need to switch the shutter anyway, then switch to OPTICAL_PUMPING_TA, repump is off using shutter.